STACpopulator/logging.py

```diff
@@ -0,0 +1,128 @@
+import datetime as dt
+import json
+import logging
+
+LOG_RECORD_BUILTIN_ATTRS = {
+    "args",
+    "asctime",
+    "created",
+    "exc_info",
+    "exc_text",
+    "filename",
+    "funcName",
+    "levelname",
+    "levelno",
+    "lineno",
+    "module",
+    "msecs",
+    "message",
+    "msg",
+    "name",
+    "pathname",
+    "process",
+    "processName",
+    "relativeCreated",
+    "stack_info",
+    "thread",
+    "threadName",
+    "taskName",
+}
+
+
+def setup_logging(logfname: str, log_level: str) -> None:
+    """Setup the logger for the app.
+
+    :param logfname: name of the file to which to write log outputs
+    :type logfname: str
+    :param log_level: base logging level (e.g. "INFO")
+    :type log_level: str
+    """
+    config = logconfig
+    config["handlers"]["file"]["filename"] = logfname
+    for handler in config["handlers"]:
+        config["handlers"][handler]["level"] = log_level
+    logging.config.dictConfig(config)
+
+
+class JSONLogFormatter(logging.Formatter):
+    # From: https://github.com/mCodingLLC/VideosSampleCode/tree/master/videos/135_modern_logging
+    def __init__(
+        self,
+        *,
+        fmt_keys: dict[str, str] | None = None,
+    ):
+        super().__init__()
+        self.fmt_keys = fmt_keys if fmt_keys is not None else {}
+
+    def format(self, record: logging.LogRecord) -> str:
+        message = self._prepare_log_dict(record)
+        return json.dumps(message, default=str)
+
+    def _prepare_log_dict(self, record: logging.LogRecord):
+        always_fields = {
+            "message": record.getMessage(),
+            "timestamp": dt.datetime.fromtimestamp(record.created, tz=dt.timezone.utc).isoformat(),
+        }
+        if record.exc_info is not None:
+            always_fields["exc_info"] = self.formatException(record.exc_info)
+
+        if record.stack_info is not None:
+            always_fields["stack_info"] = self.formatStack(record.stack_info)
+
+        message = {
+            key: msg_val if (msg_val := always_fields.pop(val, None)) is not None else getattr(record, val)
+            for key, val in self.fmt_keys.items()
+        }
+        message.update(always_fields)
+
+        for key, val in record.__dict__.items():
+            if key not in LOG_RECORD_BUILTIN_ATTRS:
+                message[key] = val
+
+        return message
+
+
+class NonErrorFilter(logging.Filter):
+    def filter(self, record: logging.LogRecord) -> bool | logging.LogRecord:
+        return record.levelno <= logging.INFO
+
+
+logconfig = {
+    "version": 1,
+    "disable_existing_loggers": False,
+    "formatters": {
+        "simple": {
+            "()": "colorlog.ColoredFormatter",
+            "format": "  %(log_color)s%(levelname)s:%(reset)s %(blue)s[%(name)-30s]%(reset)s %(message)s",
+            "datefmt": "%Y-%m-%dT%H:%M:%S%z",
+        },
+        "json": {
+            "()": "STACpopulator.logging.JSONLogFormatter",
```

Comment:
Consider `f"{JSONLogFormatter.__module__}.{JSONLogFormatter.__name__}"` so that IDEs can resolve references and make the config dynamic with the actual location.